src/app/main/time-groups/edit-time-group/edit-time-group.component.ts

```diff
@@ -212,28 +229,110 @@ export class EditTimeGroupComponent implements OnInit {
+        let data = {
+          title: this.translate.instant("tr_unassign_message"),
+          buttons: [
+            {
+              label: this.translate.instant("tr_action.cancel"),
+              actionCallback: 'cancel',
+              type: 'btn-secondary'
+            },
+            {
+              label: this.translate.instant("tr_action.ok"),
+              actionCallback: 'ok',
+              type: 'btn-primary'
+            }
+          ]
+        }
+        const dialogRef = this.dialog.open(SkoleraConfirmationComponent, {
+          width: '400px',
+          data: data,
+          disableClose: true
+        });
+        dialogRef.afterClosed().subscribe(result => {
+          if (result == 'ok') {
+            let params = {
+              "employee": {
+                "time_group_id": null
+              }
+            }
+            this.subscriptions.push(this.employeesService.updateEmployee(timegroupEmployee.id, params).subscribe((response: any) => {
+              this.appNotificationService.push(this.translate.instant('tr_unassign_time_group_successfully'), 'success');
+              timegroupEmployee.isInsideCurrentTimeGroup = true
+              this.timeGroup.employees?.push(timegroupEmployee)
+              this.filterTimeGroupEmployees()
+              this.employeesList = [];
+              this.getEmployees();
+            }, error => {
+              this.appNotificationService.push('There was an unexpected error, please reload', 'error');
+              timegroupEmployee.isInsideCurrentTimeGroup = false
+              event.target.checked = false
+            }))
+
+          }
+          else {
+            timegroupEmployee.isInsideCurrentTimeGroup = false;
+            event.target.checked = false
+            
+          }
+        })
+      }
+      else {
+        timegroupEmployee.isInsideCurrentTimeGroup = true;
+        this.timeGroup.employees?.push(timegroupEmployee)
+      }
+
     }
     else {
+      timegroupEmployee.isInsideCurrentTimeGroup = false
       this.timeGroup.employees = this.timeGroup.employees!.filter(employee => employee.name != timegroupEmployee.name)
+      this.filteredTimeGroupEmployees = this.filteredTimeGroupEmployees!.filter(employee => employee.name != timegroupEmployee.name)
     }
+    this.timeGroupEmployees = JSON.parse(JSON.stringify(this.timeGroup.employees))
   }
 
   public filterEmployees(term: any, searchKey: string) {
-    term = (searchKey == 'by_department_id') ? term : term.target.value
-    this.employessParams[searchKey] = term;
+
+    if(!term || (searchKey != 'by_department_id' && (term == '' || term.target.value == ''))){
+      delete this.employessParams[searchKey]
+    }else{
+      const searchTerm = (searchKey == 'by_department_id') ? term.id : term.target.value
+      this.employessParams[searchKey] = searchTerm;
+    }
     this.employeesList = [];
-    this.getEmployees();
+    this.getEmployees('search');
```

Comment:
You don't need this `this.employeesList = [];`

Reply:
removed